api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/ClaimService.java

```diff
@@ -34,33 +37,46 @@
     );
 
     @SuppressWarnings("PMD.AvoidCatchingGenericException")
-    public Claimant createClaim(ClaimDTO claimDTO) {
+    public Claim createClaim(ClaimDTO claimDTO) {
         Claim claim = converter.convert(claimDTO);
-
         Claimant claimant = claim.getClaimant();
 
         try {
-            EligibilityStatus eligibilityStatus = determineEligibilityStatus(claimant);
+            EligibilityResponse eligibilityResponse = determineEligibilityStatus(claimant);
+            EligibilityStatus eligibilityStatus = eligibilityResponse.getEligibilityStatus();
             ClaimStatus claimStatus = STATUS_MAP.get(eligibilityStatus);
+            updateClaimFromEligibilityResponse(claim, eligibilityResponse);
             saveClaimant(claimant, claimStatus, eligibilityStatus);
-            return claimant;
+            return claim;
         } catch (RuntimeException e) {
             saveClaimant(claimant, ClaimStatus.ERROR, EligibilityStatus.ERROR);
             throw e;
         }
     }
 
-    private EligibilityStatus determineEligibilityStatus(Claimant claimant) {
+    private EligibilityResponse determineEligibilityStatus(Claimant claimant) {
         if (claimantRepository.eligibleClaimExistsForNino(claimant.getNino())) {
-            return EligibilityStatus.DUPLICATE;
+            return EligibilityResponse.builder()
+                    .eligibilityStatus(EligibilityStatus.DUPLICATE)
+                    .build();
         } else {
             EligibilityResponse eligibilityResponse = client.checkEligibility(claimant);
-            claimant.setDwpHouseholdIdentifier(eligibilityResponse.getDwpHouseholdIdentifier());
-            claimant.setHmrcHouseholdIdentifier(eligibilityResponse.getHmrcHouseholdIdentifier());
-            return eligibilityStatusCalculator.determineEligibilityStatus(eligibilityResponse);
+            EligibilityStatus eligibilityStatus = eligibilityStatusCalculator.determineEligibilityStatus(eligibilityResponse);
+            return eligibilityResponse.toBuilder()
+                    .eligibilityStatus(eligibilityStatus)
+                    .build();
         }
     }
 
+    private void updateClaimFromEligibilityResponse(Claim claim, EligibilityResponse eligibilityResponse) {
```

Comment:
If you have to mutate an object from a method, I always like to return the mutated object from that method, it makes it slightly clearer that you are doing so.